libopenage/curve/tests/curve_types.cpp

```diff
@@ -386,6 +387,86 @@ void curve_types() {
 		TESTEQUALS(c.get(1), 0);
 		TESTEQUALS(c.get(5), 0);
 	}
+
+	{ // array
+		auto f = std::make_shared<event::EventLoop>();
+
+		Array<int, 4> a(f,0);
+		a.set_insert(time::time_t(1), 0, 0);
+		a.set_insert(time::time_t(1), 1, 1);
+		a.set_insert(time::time_t(1), 2, 2);
+		a.set_insert(time::time_t(1), 3, 3);
+		//a = [[0:0, 1:0],[0:0, 1:1],[0:0, 1:2],[0:0, 1:3]]
+
+		auto res = a.get_all(time::time_t(1));
+		TESTEQUALS(res[0], 0);
```

Comment:
You should use `at()` for array access in test code, since it throws errors when the index is not in range. This will save us from unexpected surprises with faulty test code.

```suggestion
		TESTEQUALS(res.at(0), 0);
```